### PR TITLE
Fix resources being skipped in InstancePlaceholder

### LINKED ISSUE #94266: InstancePlaceholder doesn't assign Resource to instance
### Tested versions

These versions work fine:
v4.2.2.stable.official [15073afe3]
v4.3.dev1.official [9d1cbab1c]
v4.3.dev2.official [352434668]
v4.3.dev3.official [36e943b6b]
v4.3.dev4.official [df78c0636]
v4.3.dev5.official [89f70e98d]
v4.3.dev6.official [89850d553]

Bug appears with:
v4.3.beta1.official [a4f2ea91a]

### System information

Windows 11 - v4.3.beta3.official [82cedc83c]

### Issue description

Node instantiated via InstancePlaceholder is missing Resource assigned in editor. 

### Steps to reproduce

A node that is a scene root has an export var that defaults to null such as
`@export var res:Resource`
That scene is placed in the editor in another scene, marked 'Load as Placeholder', and there we assign a new resource to 'res'.
After using create_instance on it at runtime, we observe the assigned resource is null.

### Minimal reproduction project (MRP)

[instanceplaceholder_bug.zip](https://github.com/user-attachments/files/16194922/instanceplaceholder_bug.zip)

Run and read console.

### PR DESCRIPTION
**Pull request by matheusmdx**
Fix #94266

image

<br>

I also tested with mrp from https://github.com/godotengine/godot/issues/88662#issuecomment-2037487372 and seems to be working as expected

image

### CODE REVIEW

File: scene/main/instance_placeholder.cpp [hunk 1]
```diff
@@ -160,13 +160,13 @@ void InstancePlaceholder::set_value_on_instance(InstancePlaceholder *p_placehold
 		// but the instance wants a Node Array.
 	}
 
+	Ref<Resource> resource = p_set.value;
```
Comment (akien-mga) [inline scene/main/instance_placeholder.cpp:163]: This should be in the `case Variant::Type::NIL`, since it's only used there.

Comment (matheusmdx) [reply scene/main/instance_placeholder.cpp]: Fixed.

I tried to do that at the first but i received an error about jumping variable initialization so i thought that wasn't allowed (i have little knowledge about c++) but i did a fast research and i just need to enclose the case with curly brackets.

File: scene/main/instance_placeholder.cpp [hunk 2]
```diff
@@ -161,19 +161,20 @@ void InstancePlaceholder::set_value_on_instance(InstancePlaceholder *p_placehold
...
+			// If it's nil but we have a NodePath or a Resource, we guess what works.
 			p_instance->set(p_set.name, p_set.value, &is_valid);
 			if (is_valid) {
 				break;
 			}
 
 			p_instance->set(p_set.name, try_get_node(p_placeholder, p_instance, p_set.value), &is_valid);
 			break;
+		}
 		case Variant::Type::OBJECT:
```
Comment (akien-mga) [inline scene/main/instance_placeholder.cpp:178]: Could you add braces for this one too while at it, for consistency?

Comment (matheusmdx) [reply scene/main/instance_placeholder.cpp:178]: No problem. I also should encloses the default or that is unnecessary?

Comment (akien-mga) [reply scene/main/instance_placeholder.cpp:178]: Yeah let's do it too for consistency.

**Review by akien-mga — APPROVED** [review]

File: scene/main/instance_placeholder.cpp (hunk 2 above)
Comment (matheusmdx) [reply scene/main/instance_placeholder.cpp]: Done

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**